pkg/gofr/handler_test.go

```diff
@@ -99,6 +99,71 @@ func TestHandler_ServeHTTP_Panic(t *testing.T) {
 	assert.Contains(t, w.Body.String(), http.StatusText(http.StatusInternalServerError), "TestHandler_ServeHTTP_Panic Failed")
 }
 
+func TestHandler_ServeHTTP_WithHeaders(t *testing.T) {
+	testCases := []struct {
+		desc       string
+		method     string
+		data       interface{}
+		headers    map[string]string
+		err        error
+		statusCode int
+		body       string
+	}{
+		{
+			desc:   "Response with headers, method is GET, no error",
+			method: http.MethodGet,
+			data: response.Response{
+				Headers: map[string]string{
+					"X-Custom-Header": "custom-value",
+					"Content-Type":    "application/json",
+				},
+				Data: map[string]string{
+					"message": "Hello, World!",
+				},
+			},
+			headers: map[string]string{
+				"X-Custom-Header": "custom-value",
+				"Content-Type":    "application/json",
+			},
+			err:        nil,
+			statusCode: http.StatusOK,
+			body:       `{"message":"Hello, World!"}`,
+		},
+		{
+			desc:       "No headers, method is GET, data is simple string, no error",
+			method:     http.MethodGet,
+			data:       "simple string",
+			headers:    nil,
+			err:        nil,
```

Comment:
I wouldn't have set parameters to their default value

```suggestion
```

Reply:
I have removed the setting of default values explicitly.